### PR TITLE
8290731: Clean up CDS handling of LambdaForm Species classes

### PR DESCRIPTION
**Pull request by iklam**
Please see the JBS issue for the reasons for this small clean up.

The main change is in `LambdaFormInvokers::regenerate_holder_classes()` -- do not regenerate the Species_XXX classes. However, mark them as "generated" to be compatible with JVMTI.

I renamed `Klass::is_regenerated()` to `Klass::is_generated_shared_class()` for clarity.

I also updated the test case to be clear what it's trying to do -- Species_IL should be loaded from base archive and Species_JL should be loaded from the dynamic archive.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8290731](https://bugs.openjdk.org/browse/JDK-8290731): Clean up CDS handling of LambdaForm Species classes

### Reviewers
 * [Calvin Cheung](https://openjdk.org/census#ccheung) (@calvinccheung - **Reviewer**) ⚠️ Review applies to [08868fc6](https://git.openjdk.org/jdk/pull/9577/files/08868fc6e39935d5577f46faaa679a940eff13a0)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/9577/head:pull/9577` \
`$ git checkout pull/9577`

Update a local copy of the PR: \
`$ git checkout pull/9577` \
`$ git pull https://git.openjdk.org/jdk pull/9577/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 9577`

View PR using the GUI difftool: \
`$ git pr show -t 9577`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/9577.diff">https://git.openjdk.org/jdk/pull/9577.diff</a>

</details>

### CODE REVIEW

File: test/hotspot/jtreg/runtime/cds/appcds/dynamicArchive/CDSLambdaInvoker.java [hunk 1]
```diff
@@ -28,9 +28,15 @@
 
 public class CDSLambdaInvoker {
     public static void main(String args[]) throws Throwable {
+        // The following calls trigger the generation of new Species classes
+        // that are not included in the base archive (or the default modules image.
```
Comment (calvinccheung) [inline test/hotspot/jtreg/runtime/cds/appcds/dynamicArchive/CDSLambdaInvoker.java:32]: Extra open parenthesis?

**Review by calvinccheung — APPROVED** [review]
Looks good. One small nit.

**Comment by iklam** [comment]
Thanks @calvinccheung for the review.
Passed tiers 1 - 4.
/integrate

**Pull request closed without merging**